framework/include/base/MooseApp.h

```diff
@@ -616,6 +571,9 @@ class MooseApp :
 
 private:
 
+  /// pointer to the parent/owning MultiApp object if any.
+  MultiApp * _parent_multi = nullptr;
```

Comment:
This is not how MultiApps are supposed to work.  They're supposed to have information pushed _down_ into them.  They should not know or care about their parent.